src/ucp/core/ucp_context.c

```diff
@@ -973,19 +976,21 @@ ucp_is_resource_in_transports_list(const char *tl_name,
      * transport which can be used as an auxiliary.
      * E.g: UCX_TLS=^tl_name,tl_name:aux, or alias={tl_name} and
      * UCX_TLS=^alias,alias:aux. */
-    if (((ret & UCP_TL_IN_TRANSPORTS_LIST_REGULAR) != 0) &&
-        ((ret & UCP_TL_IN_TRANSPORTS_LIST_AUX_IN_MAIN) != 0)) {
+    if (ucs_test_all_flags(search_result,
+                           UCP_TRANSPORTS_LIST_SEARCH_RESULT_PRIMARY |
+                           UCP_TRANSPORTS_LIST_SEARCH_RESULT_AUX_IN_MAIN)) {
         return 0;
     }
 
     /* If alias1={tl_name1, tl_name2:aux} and alias2={tl_name2}, then
      * UCX_TLS=^tl_name2 or UCX_TLS=^alias2 should not break the usage of
-     * tl_name1. */
-    if ((ret & UCP_TL_IN_TRANSPORTS_LIST_REGULAR) != 0) {
+     * tl_name1 keeping tl_name2 enabled as auxiliary transport. */
+    if (search_result & UCP_TRANSPORTS_LIST_SEARCH_RESULT_PRIMARY) {
```

Comment:
why do we need to set *rsc_flags to UCP_TL_RSC_FLAG_AUX when we found a primary transport?
is it because it's an alias of another transport that was not disabled?
also what if in this example, `UCX_TLS=^alias1,tl_name2`?

Reply:
E.g.: TLS_UCX=^ud -> UD transports are still enabled as auxiliary transports.

`UCX_TLS=^alias1,tl_name2` -> tl_name1 is disabled, tl_name2 is enabled as auxiliary. tl_name2 will be compeltely disabled, if: `UCX_TLS=^tl_name2,tl_name2:aux`,  `UCX_TLS=^alias2,tl_name2:aux`, `UCX_TLS=^alias2:aux,tl_name2`